### PR TITLE
[DRAFT] Remove redundant ProjectNode's in PredicatePushdown

### PR DESCRIPTION
**Pull request by aaneja**
Possbile fixes for #19749

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java [hunk 1]
```diff
@@ -67,6 +67,7 @@
 import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.Iterables;
 import io.airlift.slice.Slices;
+import org.jetbrains.annotations.NotNull;
```
Comment (aaneja) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java:70]: Junk, will remove

**Comment by aaneja** [comment]
Temporarily switch to 'ahana' branch for merge target to do verifier runs.  
TPCH SF 100 Parquet run - https://oss.ahana.dev/verifier-runs/59. This validated that we don't see issues with the [ProjectNode](https://github.com/prestodb/presto/pull/19766/commits/127b5ff182c767634b325bad45b056aaf3af7836) removal

**Comment by yingsu00** [comment]
> Temporarily switch to 'ahana' branch for merge target to do verifier runs. TPCH SF 100 Parquet run - https://oss.ahana.dev/verifier-runs/59. This validated that we don't see issues with the [ProjectNode](https://github.com/prestodb/presto/pull/19766/commits/127b5ff182c767634b325bad45b056aaf3af7836) removal

Can you also verify TPCDS please?

File: presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdown.java [hunk 2]
```diff
@@ -410,8 +410,8 @@ public PlanNode visitFilter(FilterNode filter, RewriteContext<Void> context)
                     tableScan.getOutputVariables(),
                     tableScan.getAssignments(),
                     tableScan.getTableConstraints(),
-                    layout.getPredicate(),
-                    TupleDomain.all());
+                    TupleDomain.all(),
```
[the remark this answers is not in the document]
Comment (aaneja) [reply presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdown.java:413]: The actual predicate used for stats computation will use the `hiveLayoutHandle.getDomainPredicate()`  + the `currentConstraint` ([link](https://github.com/prestodb/presto/blob/82441f5ef3d6f38dc61428db20039d9dc95dae1f/presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java#L876-L878)) that would've already been pushed down. So IMO, no change is needed to the `TableScanStatsRule`

**Pull request closed without merging**